src/lib/db/project-store.ts

```diff
@@ -1,23 +1,48 @@
+import { Knex } from 'knex';
+import { Logger, LogProvider } from '../logger';
+
 const NotFoundError = require('../error/notfound-error');
 
 const COLUMNS = ['id', 'name', 'description', 'created_at'];
 const TABLE = 'projects';
 
+export interface IProject {
+    id: number;
+    name: string;
+    description: string;
+    createdAt: Date;
+}
+
+interface IProjectInsert {
+    id: number;
+    name: string;
+    description: string;
+}
+
+interface IProjectArchived {
+    id: number;
+    archived: boolean;
+}
+
 class ProjectStore {
-    constructor(db, getLogger) {
+    private db: Knex;
+
+    private logger: Logger;
+
+    constructor(db: Knex, getLogger: LogProvider) {
         this.db = db;
         this.logger = getLogger('project-store.js');
     }
 
-    fieldToRow(data) {
+    fieldToRow(data): IProjectInsert {
```

Comment:
What's the type of `data`? Bit surprised that typescript didn't warn me that data was untyped here.